macros/udfs/business_hours.sql

```diff
@@ -0,0 +1,61 @@
+{% macro create_udf_business_hours() %}
+
+create or replace function {{target.schema}}.udf_business_hours(start_datetime TIMESTAMP_NTZ, end_datetime TIMESTAMP_NTZ, country STRING)
+returns number
+language python
+runtime_version = 3.8
+packages = ('numpy','holidays')
+handler = 'business_hours_py'
+as
+
+$$
+from datetime import datetime
+from datetime import timedelta
+import holidays
+import numpy as np
+
+def business_hours_py(start_datetime, end_datetime, country='US'):
+
+    # Year range for holidays spanning the years below
+    years = [*range(1990,2030)]
+    
+    # Hard coded business hours, opening and closing times
+    workhours_per_day = 8
+    opening_hour = 9
+    closing_hour = 17
+
+    # Create open and closing datetimes to establish day 0 and day n durations
+    first_day_closed = start_datetime.replace(hour=closing_hour, minute=0, second=0)
+    last_day_opened = end_datetime.replace(hour=opening_hour, minute=0, second=0)
+
+    holiday_list = list(holidays.country_holidays(country, years=years))
+    days = np.busday_count(
+        start_datetime.date(),
+        end_datetime.date(),
+        holidays = holiday_list)
+    
+    # Calculate hours for full business days (excluding first day and last day)
+    complete_days_hours = (days - 2) * workhours_per_day
+
+    # Jump forward to the next non-holiday weekday
+    while start_datetime.date() in holiday_list or start_datetime.weekday() in [5,6]: 
+        start_datetime = start_datetime.replace(
+            hour = opening_hour, minute = 0, second = 0
+            ) + timedelta(days = 1)
```

Comment:
need this step here, cannot use logic from first_day_closed on L28 because you need to loop through this until you fall upon the first non-weekend, non-holiday day and determine that day at opening_hour as your start point.